### PR TITLE
feat: generate group merkle proof

### PR DESCRIPTION
**Pull request by 0xmad**
## Explanation

Details are below:
- [x] Add group merkle proof page
- [x] Add events for proof generation

## Related Issues

Related to #316 

## Screenshots

<details>
<summary>Expand</summary>

</details>

## Manual Testing Steps

N/A

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] Pre-commit and pre-push hook checks are passed
- [x] E2E tests are passed locally
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

File: packages/providers/src/event/types.ts [hunk 1]
```diff
@@ -37,18 +39,3 @@ export enum EventName {
...
-/**
- * Enumeration representing possible rejected requests.
- *
- * @enum {string}
- * @readonly
- * @property {string} ADD_VERIFIABLE_CREDENTIAL - "addVerifiableCredential"
- * @property {string} REVEAL_COMMITMENT - "revealCommitment"
- * @property {string} JOIN_GROUP - "joinGroup"
- */
-export enum RejectRequests {
```
Comment (0xisk) [inline packages/providers/src/event/types.ts:50]: That's even better to only use EventNames enum

File: packages/e2e/playwright.config.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export default defineConfig({
   expect: {
     timeout: 15_000,
   },
-  maxFailures: 0,
```
Comment (0xisk) [inline packages/e2e/playwright.config.ts:17]: Why this change?

I mean I actually prefer it to be like this `1` so that if one e2e test failed the workflow stops until get fixed. What do you think?

File: packages/app/src/background/services/bandada/BandadaService.ts [hunk 3]
```diff
@@ -58,9 +58,16 @@ export class BandadaService {
...
       headers: DEFAULT_HEADERS,
-    }).then((res) => res.json() as Promise<string>);
+    });
+
+    if (!response.ok) {
+      const result = (await response.json()) as { message: string };
+      throw new Error(result.message.toString());
+    }
+
+    const result = (await response.json()) as string;
```
Comment (0xisk) [inline packages/app/src/background/services/bandada/BandadaService.ts:68]:
```suggestion
    const result = (await response.json()) as unknown;

    if (!response.ok) {
      throw new Error((result as { message: string }).message);
    }

    return JSON.parse(result as string) as boolean;
```

**Review by 0xisk — CHANGES_REQUESTED** [review]
Thanks @0xmad!

File: packages/e2e/playwright.config.ts (hunk 2 above)
Comment (0xmad) [reply packages/e2e/playwright.config.ts:17]: It's actually 1 because 0 disables the behaviour you described above.

File: packages/app/src/background/services/bandada/BandadaService.ts (hunk 3 above)
Comment (0xmad) [reply packages/app/src/background/services/bandada/BandadaService.ts:68]: I'll fix this in this PR https://github.com/CryptKeeperZK/crypt-keeper-extension/pull/873

**Review by 0xisk — APPROVED** [review]
Thanks @0xmad!
